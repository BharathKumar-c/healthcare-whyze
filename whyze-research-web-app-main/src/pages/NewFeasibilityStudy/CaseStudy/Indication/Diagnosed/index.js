import React from 'react';
import PropTypes from 'prop-types';
import { Row, Space, Tag } from 'antd';
import {
  CustomSelect,
  Image,
  Input,
  Select,
  Text,
} from '../../../../../components/basic';
import { ConditionContainer } from '../../../../../components/shared';
import { conditionOptions, periodOptions } from '../../../../../constant';
import DragIcon from '../../../../../assets/DragIcon.svg';
import AddIcon from '../../../../../assets/AddIcon.svg';
import CloseIcon from '../../../../../assets/CloseIcon.svg';
import { Vector } from '../../../../../assets';
import Indication from '..';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import { useDispatch, useSelector } from 'react-redux';
import {
  addChildInCaseStudy,
  removeChildInCaseStudy,
  updateCaseStudy,
  updateConditionChanges,
} from '../../../../../components/shared/AddCondition';
import { v4 as uuid } from 'uuid';
import { totalPatientFormatter } from '../../../../../utils/calculations';
import useNewFeasibilityStudyService from '../../../useNewFeasibilityStudyService';

export default function Diagnosed({
  totalpatient,
  caseDetails,
  parentIndex,
  parentId,
  isChild,
  operationValue,
  handleAddCondition,
  handleChangeCaseName,
  getPopupContainer,
}) {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const { optimusValue, isDisableInput } = useNewFeasibilityStudyService();

  const dispatch = useDispatch();
  const handleAddNewCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    addChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      {
        id: uuid().slice(0, 8),
        condition: '',
        operation: 'And',
        child: [],
      },
    );
    dispatch(setSegments(study));
  };

  const handleRemoveCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    removeChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      parentId,
      id,
    );
    dispatch(setSegments(study));
  };

  const handleSelectRelation = (value, id) => {
    const study = JSON.parse(JSON.stringify(segments));
    const newValue = conditionOptions.find((a) => a.key === value).label;
    updateCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      'operation',
      newValue,
    );
    optimusValue({
      id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  const handleInputChange = (event) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'diagnosed',
      {
        period: caseDetails?.diagnosed?.period,
        value: event?.target?.value,
      },
    );
    optimusValue({
      id: caseDetails?.id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  const handleSelect = (value) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'diagnosed',
      { value: caseDetails?.diagnosed?.value, period: value },
    );
    optimusValue({
      id: caseDetails?.id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  return (
    <ConditionContainer
      className={'condition-container'}
      isActive
      isVertical
      isChild={isChild}
    >
      <Space direction="vertical" className="container">
        <div className="container-alignment">
          <div className="drag-drop-image">
            <Image
              src={DragIcon}
              width={10}
              height={16}
              preview={false}
              alt="drag-icon"
            />
          </div>
          <div className="add-image">
            <Image
              className="image"
              src={AddIcon}
              alt="add-icon"
              width={12}
              height={12}
              preview={false}
              onClick={() => handleAddNewCondition(caseDetails.id)}
            />
          </div>
          <div className="condition-container-before-select" />
          <CustomSelect
            className="condition-container-custom-select"
            options={conditionOptions}
            onSelect={(value) => handleSelectRelation(value, caseDetails.id)}
            defaultValue={operationValue}
            getPopupContainer={getPopupContainer}
          />
          <div className="condition-container-after-select" />
          <Text className="indication-text" text="Diagnosed at least" />
          <div className="condition-container-input">
            <Input
              type="number"
              lable="Enter value"
              name="diagnosed"
              value={caseDetails?.diagnosed?.value}
              handleInputChange={handleInputChange}
              disabled={isDisableInput}
              onWheel={(e) => e.target.blur()}
            />
          </div>
          <div
            className="condition-container-select"
            id="condition-container-select"
          >
            <Select
              lable="Select type"
              ariaLabel="diagnosed-level"
              options={periodOptions}
              name="period"
              getPopupContainer={() =>
                document.getElementById('condition-container-select')
              }
              value={caseDetails?.diagnosed?.period}
              handleSelect={handleSelect}
            />
          </div>
          <Text className="indication-text-label" text="ago" />
          <div className="close-icon">
            <Image
              src={CloseIcon}
              className="image"
              alt="close-icon"
              width={20}
              height={20}
              preview={false}
              onClick={() => handleRemoveCondition(caseDetails.id, parentId)}
            />
          </div>
        </div>
        <Row className="indication-no-patient-tag-row">
          <Tag className="indication-no-patient-tag">
            <Vector />
            {totalPatientFormatter(totalpatient || 0)}
          </Tag>
        </Row>
      </Space>
      {caseDetails?.child?.length > 0 && (
        <Indication
          isChild
          parentIndex={parentIndex}
          caseDetails={caseDetails}
          handleAddCondition={handleAddCondition}
          handleChangeCaseName={handleChangeCaseName}
        />
      )}
    </ConditionContainer>
  );
}

Diagnosed.propTypes = {
  totalpatient: PropTypes.string,
  caseDetails: PropTypes.object,
  parentId: PropTypes.string,
  getPopupContainer: PropTypes.string,
  operationValue: PropTypes.string,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Diagnosed.defalutProps = {
  totalpatient: '',
  operationValue: '',
  parentId: '',
  getPopupContainer: '',
  parentIndex: 0,
  isChild: false,
  caseDetails: null,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
