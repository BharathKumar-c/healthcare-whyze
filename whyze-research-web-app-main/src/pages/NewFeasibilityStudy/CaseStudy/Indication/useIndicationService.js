import { useCallback, useState } from 'react';
import { feasibilityStudyApiUrls } from '../../../../constant/ApiEndPoints';
import ApiUtil from '../../../../utils/ApiUtils';
import { debounce } from '../../../../utils/CommonFunction';

export default function useIndication() {
  const [condition, setCondition] = useState([]);

  const getAllMedicationCondition = async (data) => {
    try {
      await ApiUtil.getData(
        `${feasibilityStudyApiUrls.medicalConditionSearchUrl}?value=${data}`,
      ).then((res) => {
        const { data } = res.data;
        setCondition(
          data?.elements.map((ele, index) => ({
            value: ele?.name,
            id: index,
          })),
        );
      });
    } catch (error) {
      console.log(error);
    }
  };
  const getMedicalCondition = useCallback(
    debounce(getAllMedicationCondition, 300),
    [],
  );
  return {
    condition,
    getMedicalCondition,
  };
}
