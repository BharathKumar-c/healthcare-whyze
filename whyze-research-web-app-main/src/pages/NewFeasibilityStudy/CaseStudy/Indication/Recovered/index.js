import React from 'react';
import PropTypes from 'prop-types';
import { Row, Space, Tag } from 'antd';
import {
  CustomSelect,
  Image,
  Input,
  Select,
  Text,
} from '../../../../../components/basic';
import { ConditionContainer } from '../../../../../components/shared';
import { conditionOptions, periodOptions } from '../../../../../constant';
import DragIcon from '../../../../../assets/DragIcon.svg';
import AddIcon from '../../../../../assets/AddIcon.svg';
import CloseIcon from '../../../../../assets/CloseIcon.svg';
import { Vector } from '../../../../../assets';
import Indication from '..';
import { useSelector, useDispatch } from 'react-redux';
import {
  addChildInCaseStudy,
  removeChildInCaseStudy,
  updateCaseStudy,
  updateConditionChanges,
} from '../../../../../components/shared/AddCondition';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import { v4 as uuid } from 'uuid';
import { totalPatientFormatter } from '../../../../../utils/calculations';
import useNewFeasibilityStudyService from '../../../useNewFeasibilityStudyService';

export default function Recovered({
  totalPatient,
  parentIndex,
  caseDetails,
  isChild,
  parentId,
  operationValue,
  handleAddCondition,
  handleChangeCaseName,
  getPopupContainer,
}) {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();
  const { optimusValue, isDisableInput } = useNewFeasibilityStudyService();

  const handleAddNewCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    addChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      {
        id: uuid().slice(0, 8),
        condition: '',
        operation: 'And',
        child: [],
      },
    );
    dispatch(setSegments(study));
  };

  const handleSelectRelation = (value, id) => {
    const study = JSON.parse(JSON.stringify(segments));
    const newValue = conditionOptions.find((a) => a.key === value).label;
    updateCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      'operation',
      newValue,
    );
    optimusValue({
      id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  const handleRemoveCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    removeChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      parentId,
      id,
    );
    dispatch(setSegments(study));
  };

  const handleInputChange = (event) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'recovered',
      {
        period: caseDetails?.recovered?.period,
        value: event?.target?.value,
      },
    );
    optimusValue({
      id: caseDetails.id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  const handleSelect = (value) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'recovered',
      { value: caseDetails?.recovered?.value, period: value },
    );
    optimusValue({
      id: caseDetails.id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  return (
    <ConditionContainer
      className={'condition-container'}
      isActive
      isVertical
      isChild={isChild}
    >
      <Space direction="vertical" className="container">
        <div className="container-alignment">
          <div className="drag-drop-image">
            <Image
              src={DragIcon}
              alt="drag-icon"
              width={10}
              height={16}
              preview={false}
            />
          </div>
          <div className="add-image">
            <Image
              className="image"
              src={AddIcon}
              alt="add-icon"
              width={12}
              height={12}
              preview={false}
              onClick={() => handleAddNewCondition(caseDetails.id)}
            />
          </div>
          <div className="condition-container-before-select" />
          <CustomSelect
            className="condition-container-custom-select"
            defaultValue={operationValue}
            options={conditionOptions}
            onSelect={(value) => handleSelectRelation(value, caseDetails.id)}
            getPopupContainer={getPopupContainer}
          />
          <div className="condition-container-after-select" />
          <Text className="indication-text" text="Recovered at least" />
          <div className="condition-container-input">
            <Input
              type="number"
              lable="Enter value"
              name="Recovered"
              disabled={isDisableInput}
              value={caseDetails?.recovered?.value}
              handleInputChange={handleInputChange}
              onWheel={(e) => e.target.blur()}
            />
          </div>
          <div
            className="condition-container-select"
            id="condition-container-select"
          >
            <Select
              lable="Select type"
              options={periodOptions}
              getPopupContainer={() =>
                document.getElementById('condition-container-select')
              }
              name="period"
              ariaLabel="recovered-level"
              value={caseDetails?.recovered?.period}
              handleSelect={handleSelect}
            />
          </div>
          <Text className="indication-text-label" text="ago" />
          <div className="close-icon">
            <Image
              src={CloseIcon}
              className="image"
              alt="close-icon"
              width={20}
              height={20}
              preview={false}
              onClick={() => handleRemoveCondition(caseDetails.id, parentId)}
            />
          </div>
        </div>
        <Row className="indication-no-patient-tag-row">
          {totalPatient ? (
            <Tag className="indication-no-patient-tag">
              <Vector />
              {totalPatientFormatter(totalPatient || 0)}
            </Tag>
          ) : null}
        </Row>
      </Space>
      {caseDetails?.child?.length > 0 && (
        <>
          <Indication
            isChild
            parentIndex={parentIndex}
            caseDetails={caseDetails}
            handleAddCondition={handleAddCondition}
            handleChangeCaseName={handleChangeCaseName}
          />
        </>
      )}
    </ConditionContainer>
  );
}

Recovered.propTypes = {
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  totalPatient: PropTypes.string,
  getPopupContainer: PropTypes.string,
  operationValue: PropTypes.string,
  parentId: PropTypes.string,
  caseDetails: PropTypes.object,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Recovered.defalutProps = {
  parentIndex: 0,
  isChild: false,
  totalPatient: '',
  operationValue: '',
  getPopupContainer: '',
  parentId: '',
  caseDetails: null,
  handleAddCondition: null,
  handleChangeCaseName: null,
};
