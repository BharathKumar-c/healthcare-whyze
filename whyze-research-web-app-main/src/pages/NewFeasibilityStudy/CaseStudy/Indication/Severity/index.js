import React from 'react';
import PropTypes from 'prop-types';
import { Row, Space, Tag } from 'antd';
import {
  CustomSelect,
  Image,
  Select,
  Text,
} from '../../../../../components/basic';
import { ConditionContainer } from '../../../../../components/shared';
import {
  conditionOptions,
  severityStageOptions,
} from '../../../../../constant';
import DragIcon from '../../../../../assets/DragIcon.svg';
import AddIcon from '../../../../../assets/AddIcon.svg';
import CloseIcon from '../../../../../assets/CloseIcon.svg';
import { Vector } from '../../../../../assets';
import Indication from '..';
import { useSelector, useDispatch } from 'react-redux';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import {
  addChildInCaseStudy,
  removeChildInCaseStudy,
  updateCaseStudy,
  updateConditionChanges,
} from '../../../../../components/shared/AddCondition';
import { v4 as uuid } from 'uuid';
import { totalPatientFormatter } from '../../../../../utils/calculations';
import useNewFeasibilityStudyService from '../../../useNewFeasibilityStudyService';

export default function Severity({
  totalPatient,
  parentIndex,
  caseDetails,
  isChild,
  parentId,
  operationValue,
  handleAddCondition,
  handleChangeCaseName,
  getPopupContainer,
}) {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();

  const { optimusValue } = useNewFeasibilityStudyService();

  const handleAddNewCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    addChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      {
        id: uuid().slice(0, 8),
        condition: '',
        operation: 'And',
        child: [],
        severity: { stage: 'stage 1' },
      },
    );
    dispatch(setSegments(study));
  };

  const handleRemoveCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    removeChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      parentId,
      id,
    );
    dispatch(setSegments(study));
  };

  const handleSelectRelation = (value, id) => {
    const study = JSON.parse(JSON.stringify(segments));
    const newValue = conditionOptions.find((a) => a.key === value).label;
    updateCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      'operation',
      newValue,
    );
    optimusValue({
      id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  const handleSelect = (value) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'severity',
      { stage: value },
    );
    optimusValue({
      id: caseDetails.id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  return (
    <ConditionContainer
      className={'condition-container'}
      isActive
      isVertical
      isChild={isChild}
    >
      <Space direction="vertical" className="container">
        <div className="container-alignment">
          <div className="drag-drop-image">
            <Image
              src={DragIcon}
              width={10}
              height={16}
              preview={false}
              alt="drag-icon"
            />
          </div>
          <div className="add-image">
            <Image
              className="image"
              src={AddIcon}
              alt="add-icon"
              width={12}
              height={12}
              preview={false}
              onClick={() => handleAddNewCondition(caseDetails.id)}
            />
          </div>
          <div className="condition-container-before-select" />
          <CustomSelect
            className="condition-container-custom-select"
            options={conditionOptions}
            onSelect={(value) => handleSelectRelation(value, caseDetails.id)}
            defaultValue={operationValue}
            getPopupContainer={getPopupContainer}
          />
          <div className="condition-container-after-select" />
          <Text className="indication-text" text="Severity" />
          <div
            className="condition-container-select"
            id="condition-container-select"
          >
            <Select
              lable="Select type"
              value={caseDetails?.severity?.stage}
              getPopupContainer={() =>
                document.getElementById('condition-container-select')
              }
              ariaLabel="severity-level"
              name="severity"
              options={severityStageOptions}
              handleSelect={handleSelect}
              isShowSearch
            />
          </div>
          <div className="close-icon">
            <Image
              src={CloseIcon}
              className="image"
              width={20}
              height={20}
              preview={false}
              alt="close-icon"
              onClick={() => handleRemoveCondition(caseDetails.id, parentId)}
            />
          </div>
        </div>
        <Row className="indication-no-patient-tag-row">
          {totalPatient ? (
            <Tag className="indication-no-patient-tag">
              <Vector />
              {totalPatientFormatter(totalPatient || 0)}
            </Tag>
          ) : null}
        </Row>
      </Space>
      {caseDetails?.child?.length > 0 && (
        <>
          <Indication
            isChild
            parentIndex={parentIndex}
            caseDetails={caseDetails}
            handleAddCondition={handleAddCondition}
            handleChangeCaseName={handleChangeCaseName}
          />
        </>
      )}
    </ConditionContainer>
  );
}

Severity.propTypes = {
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  totalPatient: PropTypes.string,
  operationValue: PropTypes.string,
  getPopupContainer: PropTypes.string,
  parentId: PropTypes.string,
  caseDetails: PropTypes.object,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Severity.defalutProps = {
  parentIndex: 0,
  isChild: false,
  totalPatient: '',
  operationValue: '',
  parentId: '',
  getPopupContainer: '',
  caseDetails: null,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
