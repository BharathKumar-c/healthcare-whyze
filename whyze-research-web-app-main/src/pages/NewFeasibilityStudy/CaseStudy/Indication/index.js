import React, { useEffect, useState } from 'react';
import AddCondition from '../../../../components/shared/AddCondition';
import Diagnosed from './Diagnosed';
import Recovered from './Recovered';
import Severity from './Severity';
import { AutoCompleteComponent, Image } from '../../../../components/basic';
import AddIcon from '../../../../assets/AddIcon.svg';
import PropTypes from 'prop-types';
import {
  autocompleteDefaultLabel,
  autocompleteEmptyLabel,
  medicalConditionOptions,
} from '../../../../constant';
import useIndication from './useIndicationService';
import {
  conditions,
  indicationConstants,
} from '../../../../constant/ConstantTexts';
import { useSelector } from 'react-redux';
import useNewFeasibilityStudyService from '../../useNewFeasibilityStudyService';

const Indication = (props) => {
  const [isError, setIsError] = useState(false);
  const [isGetCount, setIsGetCount] = useState(false);
  const { condition, getMedicalCondition } = useIndication();
  const { optimusValue } = useNewFeasibilityStudyService();
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const handleError = () => {
    setIsError(true);
  };
  const handleSearch = (e) => {
    if (e.length >= 3) {
      getMedicalCondition(e);
    }
    props.handleChangeCaseName(e, props.parentIndex);
    setIsError(false);
  };

  const handleSelect = (e) => {
    props.handleChangeCaseName(e, props.parentIndex);
    setIsGetCount(true);
  };

  useEffect(() => {
    if (isGetCount) {
      optimusValue({
        name: props?.caseDetails?.name,
        data: segments[activeSegment.index]?.feasibilityStudy,
      });
      setIsGetCount(false);
    }
  }, [isGetCount]);

  return (
    <div className="indication" id="indication">
      {!props.isChild && (
        <>
          <AutoCompleteComponent
            id="main_textbox"
            options={
              props?.caseDetails?.caseName?.length >= 3
                ? condition.length > 0
                  ? condition
                  : autocompleteEmptyLabel
                : autocompleteDefaultLabel
            }
            label={indicationConstants.medicalConditionLabel}
            getPopupContainer={() => document.getElementById('indication')}
            placeholder={indicationConstants.medicalConditionLabel}
            handleSearch={handleSearch}
            handleSelect={handleSelect}
            value={props?.caseDetails?.caseName}
            width={'320px'}
            isError={isError}
            errorText={indicationConstants.medicalConditionErrorMessage}
          />
          <div style={{ height: '30px' }}>
            <Image
              style={{ cursor: 'pointer' }}
              preview={false}
              width={12}
              alt="Indication-add-icon"
              height={12}
              src={AddIcon}
              onClick={() =>
                props.caseDetails.caseName !== ''
                  ? props.handleAddCondition(props.parentIndex)
                  : handleError()
              }
            />
          </div>
        </>
      )}
      {props.caseDetails?.child?.map((c, index) => {
        return (
          <div key={`${index}_${c.condition}_${c.operation}`}>
            {props.isChild ? (
              <div
                className={
                  c?.condition === ''
                    ? 'vertical-line-condition'
                    : 'vertical-line'
                }
              />
            ) : (
              <div
                style={{
                  borderLeft: `2px ${
                    c.condition === '' ? 'dashed' : 'solid'
                  } #D3D6D9`,
                  height: '32px',
                }}
              />
            )}
            {c.condition === '' ? (
              <AddCondition
                caseDetails={c}
                isChild={props.isChild}
                parentId={props.caseDetails.id}
                parentIndex={props.parentIndex}
                value={c.operation}
                addConditionOptions={medicalConditionOptions}
                handleAddCondition={props.handleAddCondition}
                handleChangeCaseName={props.handleChangeCaseName}
                conditionName="condition"
                getPopupContainer={'indication'}
              />
            ) : c.condition === conditions.diagnosed ? (
              <Diagnosed
                caseDetails={c}
                isChild={props.isChild}
                parentIndex={props.parentIndex}
                parentId={props.caseDetails.id}
                operationValue={c.operation}
                totalpatient={c?.count}
                handleAddCondition={props.handleAddCondition}
                handleChangeCaseName={props.handleChangeCaseName}
                getPopupContainer={'indication'}
              />
            ) : c.condition === conditions.recovered ? (
              <Recovered
                caseDetails={c}
                isChild={props.isChild}
                parentIndex={props.parentIndex}
                parentId={props.caseDetails.id}
                operationValue={c.operation}
                totalPatient={c?.count}
                handleAddCondition={props.handleAddCondition}
                handleChangeCaseName={props.handleChangeCaseName}
                getPopupContainer={'indication'}
              />
            ) : c.condition === conditions.severity ? (
              <Severity
                caseDetails={c}
                isChild={props.isChild}
                parentIndex={props.parentIndex}
                operationValue={c.operation}
                totalPatient={c?.count}
                parentId={props.caseDetails.id}
                handleAddCondition={props.handleAddCondition}
                handleChangeCaseName={props.handleChangeCaseName}
                getPopupContainer={'indication'}
              />
            ) : (
              ''
            )}{' '}
          </div>
        );
      })}
    </div>
  );
};

Indication.propTypes = {
  caseDetails: PropTypes.object,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Indication.defaultProps = {
  caseDetails: null,
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};

export default Indication;
