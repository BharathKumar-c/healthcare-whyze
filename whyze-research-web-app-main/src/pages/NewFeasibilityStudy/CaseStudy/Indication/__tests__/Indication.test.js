import { fireEvent, render, waitFor, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import React from 'react';
import Indication from '..';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import { act } from 'react-dom/test-utils';
import ApiUtil from '../../../../../utils/ApiUtils';
import { medicationConstants } from '../../../../../constant/ConstantTexts';
function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
jest.mock('../../../../../utils/ApiUtils');
const feasibilityStudy = {
  name: 'Indication',
  case: 'Indication',
  condition: 'Inclusion',
  caseName: '',
  child: [],
};

describe('create test cases for Indication', () => {
  test('create snapshot for Indication', () => {
    const { container } = renderWithContext(<Indication />);
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(feasibilityStudy);
      store.dispatch(setSegments(segments));
    });
    expect(container).toMatchSnapshot();
  });
  test('Should render the input field correctly', async () => {
    renderWithContext(<Indication />);
    ApiUtil.getData.mockResolvedValue({
      data: {
        result: [
          [
            {
              Name: 'Covid-19',
            },
            {
              Name: 'Covid-19 with pneumonia, sars-cov-2 identified',
            },
          ],
        ],
      },
    });
    const inputField = screen.getByRole('combobox');

    expect(inputField).toBeInTheDocument();
    expect(inputField.value).toMatch('');

    fireEvent.change(inputField, {
      target: { value: 'Covi' },
    });
    expect(inputField.value).toMatch('Covi');

    await waitFor(() => {
      expect(ApiUtil.getData).toHaveBeenCalledWith(
        'search/indication?value=Covi',
      );
    });
  });
  test('Should show the Enter 3 or more Characters text when casename has less than three letters', () => {
    renderWithContext(<Indication />);

    const inputField = screen.getByRole('combobox');

    expect(inputField).toBeInTheDocument();
    expect(inputField.value).toMatch('');

    fireEvent.change(inputField, {
      target: { value: 'Co' },
    });

    const contentText = screen.getByText('Enter 3 or more Characters');
    expect(contentText).toBeInTheDocument();
  });
  test('Should render the Add icon correctly', () => {
    renderWithContext(<Indication />);

    const addIcon = screen.getByAltText('Indication-add-icon');
    expect(addIcon).toBeInTheDocument();
  });
  test('Should render error text when click add icon without caseName correctly', () => {
    renderWithContext(<Indication caseDetails={feasibilityStudy} />);

    const addIcon = screen.getByAltText('Indication-add-icon');
    fireEvent.click(addIcon);

    const errorText = screen.getByText(
      `${medicationConstants.medicationConditionErrorMessage}`,
    );
    expect(errorText).toBeInTheDocument();
  });
  test('Should render error text when click add icon with empty caseName correctly', () => {
    const caseDetailsWithEmptyCaseName = {
      name: 'Indication',
      case: 'Indication',
      condition: 'Inclusion',
      caseName: '',
      child: [],
    };
    renderWithContext(
      <Indication caseDetails={caseDetailsWithEmptyCaseName} />,
    );

    const addIcon = screen.getByAltText('Indication-add-icon');
    fireEvent.click(addIcon);

    const errorText = screen.getByText(
      medicationConstants.medicationConditionErrorMessage,
    );
    expect(errorText).toBeInTheDocument();
  });

  test('Should render when click add icon with caseName correctly', async () => {
    const caseDetailsWithCaseName = {
      name: 'Indication',
      case: 'Indication',
      condition: 'Inclusion',
      caseName: 'Covid-19',
      child: [],
    };

    const handleAddCondition = jest.fn();
    renderWithContext(
      <Indication
        caseDetails={caseDetailsWithCaseName}
        parentIndex={0}
        handleAddCondition={handleAddCondition}
      />,
    );

    const addIcon = screen.getByAltText('Indication-add-icon');
    fireEvent.click(addIcon);

    expect(handleAddCondition).toBeCalledTimes(1);
  });
  test('Should render vertical line when medication has child with condition', () => {
    const caseDetailsWithCaseName = {
      name: 'Indication',
      case: 'Indication',
      condition: 'Inclusion',
      caseName: 'Covid-19',
      child: [
        {
          id: '343fcad6',
          condition: 'Recovered',
          operation: 'And',
          child: [],
          recovered: {
            period: 'day',
            value: 22,
          },
        },
      ],
    };
    const { container } = renderWithContext(
      <Indication caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const verticalLine = container.getElementsByClassName('vertical-line');
    expect(verticalLine[0]).toBeInTheDocument();
  });

  test('Should render vertical line when medication has child without condition', () => {
    const caseDetailsWithCaseName = {
      name: 'Indication',
      case: 'Indication',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: '343fcad6',
          condition: '',
          operation: 'And',
          child: [],
        },
      ],
    };
    const { container } = renderWithContext(
      <Indication caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const verticalLine = container.getElementsByClassName(
      'vertical-line-condition',
    );
    const addButton = screen.getByText('+ Add other condition details');
    expect(verticalLine[0]).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
  });

  test('Should render dosage component when medication has dosage child component', () => {
    const caseDetailsWithCaseName = {
      name: 'Indication',
      case: 'Indication',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: '343fcad6',
          condition: 'Recovered',
          operation: 'And',
          child: [],
          recovered: {
            period: 'day',
            value: 22,
          },
        },
      ],
    };
    const { container } = renderWithContext(
      <Indication caseDetails={caseDetailsWithCaseName} isChild />,
    );
    expect(container).toBeInTheDocument();
  });
});
