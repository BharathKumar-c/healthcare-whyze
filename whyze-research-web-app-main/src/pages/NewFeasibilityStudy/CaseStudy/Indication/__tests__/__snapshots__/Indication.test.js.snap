// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create test cases for Indication create snapshot for Indication 1`] = `
<div>
  <div
    class="indication"
  >
    <div
      class="autoComplete-wrapper"
    >
      <span
        class="ant-typography input-lable"
      >
        Medical condition
      </span>
      <div
        class="ant-select ant-select-auto-complete autoComplete-wrapper-input ant-select-single ant-select-show-search"
        style="width: 320px;"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              id="rc_select_TEST_OR_SSR"
              role="combobox"
              type="search"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-placeholder"
          >
            Medical condition
          </span>
        </div>
      </div>
    </div>
    <div
      style="height: 30px;"
    >
      <div
        class="ant-image"
        style="width: 11px; height: 11px;"
      >
        <img
          alt="Indication-add-icon"
          class="ant-image-img"
          height="11"
          src="AddIcon.svg"
          style="height: 11px; cursor: pointer;"
          width="11"
        />
      </div>
    </div>
  </div>
</div>
`;
